src/backend/dist.js

```diff
@@ -522,15 +547,44 @@ module.exports = {
   fetchApiJsons: function(appFolder, apiEndpoint, socket, done) {
     const endpoint = apiEndpoint.replace(/\/$/, '');
     const appPath = distPath + '/' + appFolder;
+    var endpointType = 'github';
+
+    var jsonsUrl = {
+      'speakers': '',
+      'sponsors': '',
+      'sessions': '',
+      'tracks': '',
+      'microlocations': '',
+      'event': '',
+    };
+
+    Object.keys(jsonsUrl).forEach(function(key) {
+      jsonsUrl[key] = endpoint + '/' + key;
+    });
+
+    if(endpoint.search('eventyay') != -1) {
```

Comment:
can you explain the logic here ?

Reply:
Sure! The user can generate the event website using the API endpoint. Now, this API endpoint could be either the raw GitHub link or the new eventyay server link. The raw Github links directly return the whole content of the JSON file when accessed.

But, when the user has provided a link on the eventyay server, then the response is no longer the same because of the migration to the new JSON API model. We have to add certain query parameters to the request URL and then parse it using some software to get the same content back.

Like accessing sessions data from the GitHub URL is simple and can be simply fetched like
https://raw.githubusercontent.com/fossasia/open-event/master/sample/FOSSASIA16/sessions
But, in case the user has provided the server endpoint link, then to get the whole data as above, we have to send the query in a modified manner and then deserialize the JSON API response received
Request URL:
http://api.eventyay.com/v1/events/69/sessions?include=track,microlocation,session-type,speakers&fields[track]=id,name&fields[speaker]=id,name&fields[microlocation]=id,name&page[size]=0
After we receive a response from this URL, we have to deserialize the JSON API response back to get the desired data.

So this code checks whether the API endpoint provided by the user is the raw GitHub link or the eventyay one and accordingly construct the request URL which we have to use to get the desired content and then handle the response accordingly! Thanks!